io.github.spacingbat3.webcord.yml

```diff
@@ -0,0 +1,46 @@
+app-id: io.github.spacingbat3.webcord
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: run.sh
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=all
+modules:
+  - name: webcord
+    buildsystem: simple
+    build-options:
+    sources:
+      - type: archive
+        url: https://github.com/vega-d/WebCord_flatpak_build/releases/download/v0.0-rolling/WebCord_tarball.tar.gz
```

Comment:
so this contain `x86_64` binaries, and you still try to build for `aarch64` as well.

Any chance to actually get `aarch64` too?

Reply:
also this look like a "rolling" release.

1. I assume the unversioned tarball will change, so the build will fail (differing checksum)
2. flathub is NOT for rolling releases. So you should select a specific revision and go with that.

Reply:
it is rolling only in a sense that version number used in the link doesn't change. New tarball is automatically made and updated using Github Actions only when a release is published [here](https://github.com/SpacingBat3/WebCord/releases) 

When this flatpak manifest will have it's own repo we will setup another Github Actions to automatically recalculate and replace the checksum in the manifest. 

I have to admit I forgot about `aarch64`, for now it will be `x64` only until I figure out cross compiling of an `aarch64` tarball.

Reply:
That release doesn't have a tarball. This is why @axtloss and I setup an automated github actions build of the tarball linked in the manifest.

There's not really a chance the developer will go out of their way to provide a tarball right now as [they are now focused on reworking their AppImage chain instead](https://github.com/SpacingBat3/WebCord/issues/116#issuecomment-1144089798)

Reply:
Upstream WebCord doesn't offer any tarball builds, which is why we made the repo to build tarballs which are used in the flatpak.

Reply:
they have .deb. It's as good as anything else. You could also build from source.

Reply:
Vega has tried making a manifest to build from source, however WebCord uses electron-forge, which is currently broken in flatpak, which is why we chose to build a tarball,

Reply:
> how about using the .deb like a lot of other are doing when repackaging binaries?

I guess we could do .deb repacking, but we already have tarball building made and working so that'd be basically starting from scratch. 



> Again, since this is a rolling release without versioning it's not acceptable.

It is *not* a rolling release. A new tarball [is only triggered to be generated when there's a new tag release upstream. ](https://github.com/axtloss/WebCord/blob/a935c4482d0b0b22354552454c55d5c9061d79e4/.github/workflows/trigger.yml#L9)

Reply:
Once again, this isn't rolling release at all, the tarball is built according to the WebCord releases, the repo with the tarball is only marked as "rolling release" so that the link to the newest tarball doesn't change. The flatpak will be using point releases that are synced to WebCord upstream.

Reply:
> if it is trigger on a new release then why doesn't it clear show it's a release instead of being unversioned?

because that would change the url to the tarball to change.

Reply:
> You can setup something to automatically update the URL and sha256 on a new release.

That is, exactly the plan. We would like to use Github Actions to run a shell script to change out url and sha sum every time we build a new tarball, (which happens when a new release is out). It is currently not implemented since it is not exactly possible to attach a github workflow to a pull request

Reply:
Hm I guess you have a point, since we were already planning on automating the sha256 change we could make it change the url to the tarball too.

Reply:
(just to clarify they don't integrate with flathub, so that's kind of a problem)

Reply:
I thought we'd just modify the current build repo we have right now to release every tarball build as a new release and then have a ci in the flathub repo that gets triggered to modify the manifest and rebuild it, if that makes any sense (and would work).

Reply:
> @vega-d what do you think?

I thought that was the plan all along?

Reply:
Wasn't our plan to leave the tarball on the static url and then just update the sha256 in the manifest?

Reply:
> and then you could provide `aarch64` like the .deb exists.

I can't find a way to build an arm tarball using github actions, I think we will be leaving out arm support entirely for now

Reply:
> Wasn't our plan to leave the tarball on the static url and then just update the sha256 in the manifest?

this is exactly what I understood and why I was saying it wasn't an acceptable solution.

Reply:
btw it's not `arm` it's `aarch64`. we no longer support `arm`. Just to clarify.

Reply:
> > Wasn't our plan to leave the tarball on the static url and then just update the sha256 in the manifest?
> 
> this is exactly what I understood and why I was saying it wasn't an acceptable solution.

There was some internal miscommunication. The plan is (and was) to change out url and sha sum every release. The current tarball has a `v0.0-rolling` only because that's what the script to publish from a github actions vm defaulted to.